### ISSUE TITLE
Cropping from a specific point

### ISSUE THREAD

**Issue by fredcarle** [h2non/bimg#142]
Hi, is there a way to crop from a specific (x, y) point? The gravity argument to image.Crop is not flexible enough.

Thank you!

**Comment by Dynom** [comment]
Hi @fredcarle I don't think that this is currently possible. However, perhaps the new SmartCrop feature might help your use-case (in case you were doing this outside of Imaginary and wanted to let Imaginary know what center to crop from).

Cropping from a predefined center does seem like a useful feature though, so it has my vote to be implemented (fwiw).

**Comment by greut** [comment]
With a _Point_ could you still apply the existing gravity options? Like `North`, `West`? or Should it be new ones like `Middle`, `TopLeft`, `TopRight`. A drawing would probably help.

**Comment by Dynom** [comment]
Wouldn't specifying a gravity point make the abstract terms obsolete? I don't know the use-cases for gravity, never had a real need for it, so I'm curious to learn of a use-case that combines them.

**Comment by greut** [comment]
Once you have a point, width and height, you can consider various positions for the point within the dimension. _pixi.js_ calls it _anchor_.

Let's take those:
- Source picture: 1000x1000
- Thumbnail: 100x100
- Point: 500x500

And consider the following:
- Gravity: North (or anchor: `50%, 0%`), top left of the output is: 450x500
- Gravity: Middle (or anchor: `50%, 50%`), top left is : 450 x 450
- Gravity: BottomRight (or anchor: `100%, 100%`), top left is : `400 x 400`

The gravity or the anchor is where the given point sits within the cropped area.

That's probably not what was meant by @fredcarle so I'll let him explain.

**Comment by fredcarle** [comment]
To retake what @greut said, let say I have a 1000px X 1000px image. I have a point set at {x: 200, y: 300} from the top left corner. If I say that the crop size is 400px X 400px, then the resulting image with relation to the original image would be a sqare from {x: 200, y: 300} top left corner to {x: 600, y: 700} bottom right corner. Does this make sense?

**Comment by greut** [comment]
@fredcarle indeed. `{Gravity: TopLeft, X: 200, Y: 300, Width: 400, Height: 400}`

**Comment by greut** [comment]
Grabbing a `400x400` at the position `200, 300` and scaling it down to a `100x100`

```go
bimg.Options{
  AreaWidth: 400,
  AreaHeight: 400,
  Top: 200,
  Left: 300,
  Width: 100,
  Height: 100
}
```

**Comment by greut** [comment]
**edit** I've fixed the ordering in the above options. It really rotates, scales, and finally crops.

**Comment by fredcarle** [comment]
@greut My suggestion would be to crop and then scale the image. The reason being that it's easier to know initial image size and crop aria than it is to scale the image and then deduce the crop area.

**Comment by greut** [comment]
@fredcarle that's the way it is. See #169
